docs/source/entities-advanced.mdx

```diff
@@ -57,6 +57,99 @@ type Organization {
 }
 ```
 
+### Differing `@key`s across subgraphs
+
+An entity often has the exact same `@key` field(s) across subgraphs, but this isn't required. For example, you can define a `Product` entity shared between subgraphs, one with `sku` and `upc` as the `@key`s and one with only `upc` as the `@key` field:
+
+<CodeColumns>
+
+```graphql title="Products subgraph"
+type Product @key(fields: "sku") @key(fields: "upc") {
+  sku: ID!
+  upc: String!
+  name: String!
+  price: Int
+}
+```
+
+```graphql title="Inventory subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+To merge entities between subgraphs, the entity must have at least one shared field between subgraphs. For example, you can't merge the `Product` entity defined in the following subgraphs because they don't share any fields:
+
+<p style="margin-bottom: 0">❌</p>
+
+<CodeColumns>
+
+```graphql title="Products subgraph"
+type Product @key(fields: "sku") {
+  sku: ID!
+  name: String!
+  price: Int
+}
+```
+
+```graphql title="Inventory subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+The following would allow you to merge the `Product` entity since both subgraph schemas include the `upc` field, even if it's only the `@key` for one of them:
+
+<CodeColumns>
+
+```graphql title="Products subgraph"
+type Product @key(fields: "sku") {
+  sku: ID!
+  upc: String!
+  name: String!
+  price: Int
+}
+```
+
+```graphql title="Inventory subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+If subgraphs share any fields—for example `upc` in the preceding example—it's best to use these shared field(s) as `@keys` for optimal [query planner](./entities#the-query-plan) performance.
```

Comment:
subgraphs can share number of fields (for optimization purposes -> e.g. this is a common use case behind `@provides`) which cannot be used as keys as they do not uniquely identify the entity. Maybe rephrase it to indicate that if subgraphs share any fields that can be used to uniquely identify underlying entity, then they could be included as `@keys`. But again if there is a "better" key (e.g. single field vs composite key from multiple keys) they may never be used so...

Reply:
> if there is a "better" key (e.g. single field vs composite key from multiple keys) they may never be used so...

What is best for query planner optimization? Just to use one single key field as long as its available on all subgraphs? Or is allowing for multiple keys better?

Reply:
based on https://github.com/apollographql/federation/issues/2695 it appears that we prefer the simpler `@key`s

Reply:
Got it, thank you!